lib/backend/model/keys.go

```diff
@@ -33,57 +32,130 @@ var rawBoolType = reflect.TypeOf(rawBool(true))
 
 // Key represents a parsed datastore key.
 type Key interface {
-	// DefaultPath() returns a default stringified path for this object,
-	// suitable for use in most datastores (and used on the Felix API,
-	// for example).
-	DefaultPath() (string, error)
-	// DefaultDeletePath() returns a default stringified path for deleting
-	// this object.
-	DefaultDeletePath() (string, error)
+	defaultPath() (string, error)
+	defaultDeletePath() (string, error)
 	valueType() reflect.Type
 }
 
 // Interface used to perform datastore lookups.
 type ListInterface interface {
-	// DefaultPathRoot() returns a default stringified root path, i.e. path
+	// defaultPathRoot() returns a default stringified root path, i.e. path
 	// to the directory containing all the keys to be listed.
-	DefaultPathRoot() string
-	ParseDefaultKey(key string) Key
+	defaultPathRoot() string
+
+	// KeyFromDefaultPath parses the default path representation of the
+	// Key type for this list.  It returns nil if passed a different kind
+	// of path.
+	KeyFromDefaultPath(key string) Key
```

Comment:
Presumably the key parameter should be called path?  And perhaps this method should be private, similar to the other path related methods  wrapped by public helper functions?